File: JAVA/1233. Remove Sub-Folders from the Filesystem.java
class Solution {
    public List<String> removeSubfolders(String[] f) {
        Arrays.sort(f);
        List<String>r=new ArrayList<>();
        r.add(f[0]);
        for(int i=1;i<f.length;i++){
            String a=r.get(r.size()-1);
            a+='/';
            if(!f[i].startsWith(a))
            r.add(f[i]);
        }
        return r;
    }
}

File: C++/1233. Remove Sub-Folders from the Filesystem.cpp
class Solution {
public:
    vector<string> removeSubfolders(vector<string>& f) {
        sort(f.begin(), f.end());
        vector<string> r;
        r.push_back(f[0]);
        for (int i = 1; i < f.size(); i++) {
            string a = r.back();
            a+= '/';
            if (a!=f[i].substr(0,a.size())) {
                r.push_back(f[i]);
            }
        }
        return r;
    }
};